### ISSUE TITLE
Gulp 4.0 dependency broken

### ISSUE THREAD

**Issue by okennedy** [matrix-hacks/matrix-puppet-skype#24]
Running `npm install` on a fresh checkout fails while installing gulp#4.0:

```

> matrix-puppet-skype@1.0.0 postinstall /var/lib/matrix-synapse/matrix-puppet-skype
> cd node_modules/skype-http && npm install && npm run prepare

npm ERR! code 1
npm ERR! Command failed: git checkout 4.0
npm ERR! error: pathspec '4.0' did not match any file(s) known to git.
npm ERR! 

npm ERR! A complete log of this run can be found in:
npm ERR!     /var/lib/matrix-synapse/.npm/_logs/2019-02-24T22_44_18_340Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! matrix-puppet-skype@1.0.0 postinstall: `cd node_modules/skype-http && npm install && npm run prepare`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the matrix-puppet-skype@1.0.0 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /var/lib/matrix-synapse/.npm/_logs/2019-02-24T22_44_18_357Z-debug.log
```

**Comment by hooger** [comment]
Same here. Following a link I found this:
https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5
Hope it helps with transition.
Thank you in advance!

**Comment by hooger** [comment]
I also checked skype-http repository and I think even they use this library, so it is possible that it has to be done upstream.

**Comment by thomas-profitt** [comment]
@hooger Would you update the version of `skype-http` in package.json, to the latest, and see how the output changes?

**Comment by hooger** [comment]
After updating package.json to point to current version of skype-http (0.0.15 at the time of writing) `npm install` does not stop so early, but it crashes eventually. Log is the following:
https://horvathg.duckdns.org/nextcloud/index.php/s/SpPZ6xfJSa2ZtZg
Thank you for looking into it!

**Comment by hooger** [comment]
So I did the following: first I installed skype-http as stated in its Readme.md under "Running example" and it worked! There is an instruction to install gulp-cli which was doable. However, muppet-skype still does not install, complaining about not finding prepare script (it wants to execute cd somewhere && npm install && npm prepare).

Sooo, after that I purged my node install, cloned the original repo, checked out tag v11.13.0 build it, and installed it. I also recloned muppet-skype and no it compiles! yeaaaah! Still no joy btw, but thats the topic of another issue.

I don't know if it is really a solution or just a work around, I leave it to @thomas-profitt to close the issue.

Edit: I forget to add, that in the package.json file, I had to modify the version of skype-http to 0.0.15.
